### PR TITLE
Don't crash if a replay can't be read.

### LINKED ISSUE #12034: Crash when trying to open replays
Hey, 
I was at http://www.gamereplays.org and tried to open the replay file with OpenRA.
(with firefox from the temp location directory)
Then it crashed due to no rights to access that file.

openopenra replay crash

(Note: 
- Opening OpenRa with a loaded replay streight from the normal replay directory works fine. 
- Even, if I don't have the fitting map, I'll get an "incompatible replay" message.

 )

### PR DESCRIPTION
**Pull request by pchote**
Fixes #12034.

On a related note, our replay-failure error messages need a complete overhaul.  Doing that properly is out of scope for this release.

### CODE REVIEW

File: OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs [hunk 1]
```diff
@@ -77,7 +77,13 @@ public void StartGame(Arguments args)
...
 			{
-				var replayMeta = ReplayMetadata.Read(Launch.Replay);
+				ReplayMetadata replayMeta = null;
+				try
+				{
+					replayMeta = ReplayMetadata.Read(Launch.Replay);
+				}
+				catch { }
+
 				if (ReplayUtils.PromptConfirmReplayCompatibility(replayMeta, Game.LoadShellMap))
```
Comment (obrakmann) [inline OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs]: Hmm. If `ReplayMetadata.Read` throws, `replayMeta` will be left with a `null` value, right?  If so, `ReplayUtils.PromptConfirmReplayCompatibility` will produce an "Outdated engine" error message, which is misleading.

Comment (pchote) [inline OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs]: > On a related note, our replay-failure error messages need a complete overhaul. Doing that properly is out of scope for this release.

It gives the same "outdated engine" message if you tell it to load a random garbage (like the game binary) too.

Comment (obrakmann) [inline OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs]: Then let's change that message to a more general "Replay metadata could not be read" or something.

Comment (pchote) [inline OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs]: Ok, I guess... I just hope you don't ask for more changes when you see what that involves.

**Comment by pchote** [comment]
Done.

**Review by obrakmann — APPROVED** [review]
:+1:

**Comment by reaperrr** [comment]
On Windows, if I deny read access for an orarep file I still get a `System.UnauthorizedAccessException`. Not idea if it's even possible to work around that.

The changed error message works for incompatible replays, at least, so commit 2 works as advertised.

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/3586f3829e06186b970f329cb0e3f52d5ffb41ae...afab2dadb3e38be9bcab5764266eec174077f03e)

**Pull request merged**